app/replica.go

```diff
@@ -233,19 +234,25 @@ func startReplica(c *cli.Context) error {
 		status := s.Replica().GetCloneStatus()
 		if status != "completed" {
 			logrus.Infof("Set clone status as inProgress")
-			s.Replica().SetCloneStatus("inProgress")
-			if err = CloneReplica(s, "tcp://"+address, cloneIP, snapName); err != nil {
-				logrus.Infof("Set clone status as error")
-				s.Replica().SetCloneStatus("error")
+			if err := s.Replica().SetCloneStatus("inProgress"); err != nil {
 				return err
 			}
+			if err = CloneReplica(s, "tcp://"+address, cloneIP, snapName); err != nil {
+				logrus.Error("Error in cloning replica, found error ", err.Error())
+				logrus.Info("Setting the status of clone as error")
+				return s.Replica().SetCloneStatus("error")
+			}
 		}
 		logrus.Infof("Set clone status as Completed")
-		s.Replica().SetCloneStatus("completed")
+		if err := s.Replica().SetCloneStatus("completed"); err != nil {
+			return fmt.Errorf("Error in setting the status of clone as complete, found error %s", err.Error())
```

Comment:
return err instead of creating a new error.

Reply:
Because we don't know where this error is happening due to hierarchy of calls, which includes some system calls also.